service/history/replication/eager_namespace_refresher.go

```diff
@@ -193,6 +197,7 @@ func (e *eagerNamespaceRefresherImpl) SyncNamespaceFromSourceCluster(
 	if err != nil {
 		return nil, err
 	}
+	e.namespaceRegistry.InvalidateCacheKey(namespace.Name(resp.Info.Name), namespaceId)
```

Comment:
fyi, this only invalidate the cache for one host

Reply:
yeah, this should be enough. If other host has smaller ns version, it will trigger refresh again.